packages/react-scripts/custom/scripts/init/optional-modules/modules/getAntdModuleDependencies.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const getModuleDependenciesFactory = require('./utils/getModuleDependenciesFactory');
+
+module.exports = getModuleDependenciesFactory({
+  files: [
+    'modules/ui/components/Loader/Loader.tsx',
+    'styles/vendors.less',
+    'modules/core/modules/localization/components/AppLocalizations/AppLocalizations.tsx',
+    'modules/core/modules/localization/components/LocalizationsWithAntd/LocalizationsWithAntd.tsx',
+    'modules/core/modules/localization/config/index.ts',
+  ],
```

Comment:
- add `temp/test/src/styles/antd-override.less`
- add the conditional comment into `cra-template-typescript/template/src/styles/index.ts` to wrap the `export { default as vendors } from './vendors.less';`
- And if we won't include antd, then we don't need to have the theme variables in less syntax but only in plain JS. So to solve this we can have `colors.js` and `colors.less`. `colors.less` must be then add it the opt. module files dependencies and the opt. comment needs to be add to `cra-template-typescript/template/src/styles/theme/index.ts`

Reply:
1) Why adding this file when this file is not imported due to this conditional comment in this file?
![image](https://user-images.githubusercontent.com/7579303/144042489-7a91d2eb-67d5-4b69-8626-5ef4f9205540.png)

2) Those vendors are used in staticCSS so I hope this wont be a problem - I will try it locally together

3) I am not sure I am follow your lead - how can I choose between colors.js and colors.less when we have no negation of using the conditional comment? Or the colors.less should be imported inside the colors.js?

Reply:
1. Because this file is only relevant for antd. Otherwise, this global-style pattern should be avoided. A good practice would be to load the styles with given component
2. TBH I am not sure those comments work in less files. But it can be done without that by updating 
    - `cra-template-typescript/template/src/styles/index.ts` to:
      ```ts
      // @use-antd-module-start
      export { default as vendors } from './vendors.less';
      // @use-antd-module-end
      ```
    - Then everywhere the `vendors` are used (-> in `core/modules/fela`)
3. Sorry, my bad. You're right. This is not actually possible to achieve. Ok, I guess I am gonna have to extend the optional-module by `@not-use` type of comment -> https://github.com/AckeeCZ/create-react-app/issues/91

Reply:
1) I am not sure what do you want from me to do with that :D  `antd-override.less` is not used because it is in `vendors.less` file which is also excluded with the optional comment

2) It is not caused by the comment in less - I tried it. It is caused when that file is specified in `getAntdModuleDependencies.js` - dunno why
But yes, I will handle that the way you recommended, thanks

3) Yes, it would be very appreciated! <3

Reply:
- I did rebase against master the `less` files as dependencies are now correctly removed (I don't understand why, but you had a somehow old version of master). 
- Then I finished the setup of `getAntdModuleDependencies.files` to include: 
   ```
    'modules/ui/components/Loader/Loader.tsx',
    'styles/vendors.less',
    'styles/antd-override.less', // this must be here, otherwise it won't get removed when antd module is exclude
    'styles/index.ts',
    'modules/core/modules/fela/config/index.ts', // here we import the vendors, so cond. comments must be added
    'modules/core/modules/localization/components/Localizations/Localizations.tsx',
    'modules/core/modules/localization/config/index.ts',
    ```

Reply:
https://github.com/AckeeCZ/create-react-app/pull/87/commits/ad5ae9329c6f170216b1e1ab77d2a20034db32f4

Reply:
cool